File: PY/Luciano_Chavarria/BitWise/maxsubarrayxor.py
from sys import stdin, stdout, maxsize


def maxsubarray(arr, n):
    ans = -maxsize
    for i in range(n):
        curr_xor = 0
        for j in range(i, n):
            curr_xor = curr_xor ^ arr[j]
            ans = max(ans, curr_xor)
    return ans

arr = [8, 1, 2, 12]
n = len(arr)
stdout.write(f"maximo subarray xor es : {maxsubarray(arr,n)}")

File: Codigos estudiantes por lenguaje/C++/Edwin Villarraga/Bitwise/maxSubArrayXOR.cpp
#include<bits/stdc++.h>
#include<cstdlib>
//---------------//
using namespace std;
int maxSubArrayXOR(int arr[],int n){
    int ans=INT_MIN;
    for(int i=0;i<n;i++){
        int curr_xor=0;
        for(int j=i;j<n;j++){
            curr_xor = curr_xor ^ arr[j];
            ans = std::max(ans, curr_xor);
        }
    }
    return ans;
}
int main(){
    int arr[] = {8, 1, 2, 12};
    int n = sizeof (arr)/sizeof (arr[0]);
    cout<<"Maximo subarray XOR es "<<maxSubArrayXOR(arr,n)<<endl;
}

